controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1862,7 +1961,7 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 	return nil
 }
 
-func (r *BareMetalHostReconciler) hostUpdatePolicySetOwnerReference(info *reconcileInfo) error {
+func (r *BareMetalHostReconciler) hostUpdatePolicySetOwnerReference(info *reconcileInfo) (policy *metal3api.HostUpdatePolicy, err error) {
```

Comment:
nit: can probably come up with a more descriptive name for this function now that its responsibilities have grown.
We use `AcquireSecret()` to mean getting a Secret and setting the owner reference.

Reply:
renamed hostUpdatePolicySetOwnerReference() to acquireHostUpdatePolicy()